### PR TITLE
implement steakhut finance facet for AVAX/USDC pool

### PR DESCRIPTION
**Pull request by 0xlock**

### CODE REVIEW

File: contracts/facets/avalanche/PangolinDEXFacet.sol [hunk 1]
```diff
@@ -34,6 +34,6 @@ contract PangolinDEXFacet is UniswapV2DEXFacet {
      * Returns address of UniswapV2-like exchange
      **/
     function getExchangeIntermediaryContract() public override returns (address) {
-        return 0xdB5D94B8Ed491B058F3e74D029775A14477cF7fA;
+        return 0x286B8DecD5ED79c962b2d8F4346CD97FF0E2C352;
```
Comment (kavsky02) [inline contracts/facets/avalanche/PangolinDEXFacet.sol:37]: Commited by mistake?

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            stakingDetails.vaultAddress,
+            amount0Actual,
+            amount1Actual,
+            vaultToken.balanceOf(address(this)) - initialVaultBalance,
+            block.timestamp
+        );
+    }
+
+    /**
+     * Unstakes {UnstakingDetails.token0Address}, {UnstakingDetails.token1Address} token in the SteakHut pool
```
Comment (kavsky02) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:107]: from the SteakHut pool*

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        }));
+    }
+
+    /**
+     * Unstakes DAI.e from SteakHut AVAX/USDC pool
+     * @param liquidity amount of shares to be unstaked
+     * @param amount0Min minimum amount of AVAX to be unstaked
+     * @param amount1Min minimum amount of USDC to be unstaked
+     **/
+    function unstakeSteakHutAVAXUSDC(uint256 liquidity, uint256 amount0Min, uint256 amount1Min) external nonReentrant onlyOwner remainsSolvent {
```
Comment (kavsky02) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:53]: Can you change the `onlyOwner` modifier to `onlyOwnerOrInsolvent`? Liquidators need to be able to unwind this position if the account is insolvent.

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    /**
+     * Stakes AVAX/USDC in SteakHut AVAX/USDC pool
+     * @param amount0Desired amount of AVAX to be staked
+     * @param amount1Desired amount of USDC to be staked
+     * @param amount0Min minimum amount of AVAX to be staked
+     * @param amount1Min minimum amount of USDC to be staked
+     **/
+    function stakeSteakHutAVAXUSDC(uint256 amount0Desired, uint256 amount1Desired, uint256 amount0Min, uint256 amount1Min) external nonReentrant onlyOwner remainsSolvent {
+        _stakeTokenSteakHut(ISteakHutPool.StakingDetails({
+            token0Address: AVAX_TOKEN,
```
Comment (kavsky02) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:34]: Could you source the addresses of `token0`, `token1` and `vault` from the `TokenManager.getAssetAddress()`? In this way while using it with the `allowInactive` flag set to `false` we can pause usage of any of the tokens / vaults with the TokenManager itself.

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+    /**
+     * Unstakes DAI.e from SteakHut AVAX/USDC pool
+     * @param liquidity amount of shares to be unstaked
+     * @param amount0Min minimum amount of AVAX to be unstaked
+     * @param amount1Min minimum amount of USDC to be unstaked
+     **/
+    function unstakeSteakHutAVAXUSDC(uint256 liquidity, uint256 amount0Min, uint256 amount1Min) external nonReentrant onlyOwner remainsSolvent {
+        _unstakeTokenSteakHut(ISteakHutPool.UnstakingDetails({
+            token0Address: AVAX_TOKEN,
```
Comment (kavsky02) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:55]: Could you source the addresses of `token0`, `token1` and `vault` from the `TokenManager.getAssetAddress()`? In this way while using it with the `allowInactive` flag set to `false` we can pause usage of any of the tokens / vaults with the TokenManager itself.

File: contracts/interfaces/facets/avalanche/ISteakHutPool.sol [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        uint256 amountYMin
+    )
+        external
+        returns (uint256 shares, uint256 amountXActual, uint256 amountYActual);
+
+    function withdraw(
+        uint256 _shares
+    ) external returns (uint256 amountX, uint256 amountY);
+
+    function getUnderlyingAssets(uint256 _shares) external view returns (uint256 totalX, uint256 totalY);
```
Comment (kavsky02) [inline contracts/interfaces/facets/avalanche/ISteakHutPool.sol:19]: Not used?

File: contracts/lib/local/DeploymentConstants.sol [hunk 7]
```diff
@@ -20,11 +20,11 @@ library DeploymentConstants {
 
     address private constant _NATIVE_ADDRESS = 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
 
-    address private constant _DIAMOND_BEACON_ADDRESS = 0x02df3a3F960393F5B349E40A599FEda91a7cc1A7;
```
Comment (kavsky02) [inline contracts/lib/local/DeploymentConstants.sol:23]: Committed by mistake?

File: test/integration/avalanche/steakhut-finance.test.ts [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                MockTokenManagerArtifact,
+                []
+            ) as MockTokenManager;
+
+            await tokenManager.connect(owner).initialize(supportedAssets, lendingPools);
+            await tokenManager.connect(owner).setFactoryAddress(smartLoansFactory.address);
+
+            await tokenManager.setDebtCoverage(SteakHutAvaxUsdcLP, toWei("0.8333333333333333"));
+
+            await tokenManager.setDebtCoverageStaked(toBytes32("SH_AVAX_USDC_LP"), toWei("0.8333333333333333"));
```
Comment (kavsky02) [inline test/integration/avalanche/steakhut-finance.test.ts:104]: This should be not necessary as we are not using `stakedPositions` in this integration (was necessary for VectorFinance and vaults that didn't support `balanceOf()` method). Can we try and remove it?

File: test/integration/avalanche/steakhut-finance.test.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        it("should deploy a smart loan", async () => {
+            await smartLoansFactory.connect(owner).createLoan();
+            const loan_proxy_address = await smartLoansFactory.getLoanForOwner(owner.address);
+            loan = await ethers.getContractAt("SmartLoanGigaChadInterface", loan_proxy_address, owner);
+
+            wrappedLoan = WrapperBuilder
+                // @ts-ignore
+                .wrap(loan)
+                .usingSimpleNumericMock({
+                    mockSignersCount: 10,
```
Comment (kavsky02) [inline test/integration/avalanche/steakhut-finance.test.ts:145]: 3 is enough and cheaper

File: test/integration/avalanche/steakhut-finance.test.ts [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+                    dataPoints: MOCK_PRICES,
+                });
+
+            nonOwnerWrappedLoan = WrapperBuilder
+                // @ts-ignore
+                .wrap(loan.connect(nonOwner))
+                .usingSimpleNumericMock({
+                    mockSignersCount: 10,
```
Comment (kavsky02) [inline test/integration/avalanche/steakhut-finance.test.ts:153]: 3 is enough and cheaper

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+            vaultTokenSymbol: "SH_AVAX_USDC_LP",
+            amount0Desired: amount0Desired,
+            amount1Desired: amount1Desired,
+            amount0Min: amount0Min,
+            amount1Min: amount1Min
+        }));
+    }
+
+    /**
+     * Unstakes DAI.e from SteakHut AVAX/USDC pool
```
Comment (kavsky02) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:48]: Does it actually unstake DAI.e or is it AVAX/USDC that get unstaked?

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol (hunk 3 above)
Comment (mbare0) [inline contracts/facets/avalanche/SteakHutFinanceFacet.sol:53]: What is the strat for liquidation @kavsky02? If we want to unstake first, we should allow liquidator to do it once insolvent

File: contracts/facets/avalanche/PangolinDEXFacet.sol (hunk 1 above)
Comment (0xlock) [reply contracts/facets/avalanche/PangolinDEXFacet.sol:37]: oh, yeah

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol (hunk 4 above)
Comment (0xlock) [reply contracts/facets/avalanche/SteakHutFinanceFacet.sol:34]: okay, got it

File: test/integration/avalanche/steakhut-finance.test.ts (hunk 8 above)
Comment (0xlock) [reply test/integration/avalanche/steakhut-finance.test.ts:104]: sure!

File: contracts/lib/local/DeploymentConstants.sol (hunk 7 above)
Comment (0xlock) [reply contracts/lib/local/DeploymentConstants.sol]: yes

File: contracts/facets/avalanche/SteakHutFinanceFacet.sol (hunk 11 above)
Comment (0xlock) [reply contracts/facets/avalanche/SteakHutFinanceFacet.sol]: AVAX/USDC

File: contracts/interfaces/facets/avalanche/ISteakHutPool.sol (hunk 6 above)
Comment (0xlock) [reply contracts/interfaces/facets/avalanche/ISteakHutPool.sol]: it's not used

**Pull request merged**